### PR TITLE
utils/build/docker/golang/app: update app for v2 API changes

### PR DESCRIPTION
**Pull request by knusbaum**
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Comment by robertomonteromiguel** [comment]
I guess We can change the status of this PR to Draft

File: utils/scripts/load-binary.sh [hunk 1]
```diff
@@ -216,6 +216,12 @@ elif [ "$TARGET" = "golang" ]; then
 
     echo "Using github.com/DataDog/dd-trace-go/v2@$COMMIT_ID"
     echo "github.com/DataDog/dd-trace-go/v2@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/IBM/sarama.v1@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/gin-gonic/gin@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/go-chi/chi.v5@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/google.golang.org/grpc@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/labstack/echo.v4@$COMMIT_ID" > golang-load-from-go-get
+    echo "github.com/DataDog/dd-trace-go/v2/contrib/net/http@$COMMIT_ID" > golang-load-from-go-get
```
Comment (darccio) [inline utils/scripts/load-binary.sh:224]: @knusbaum I think we should use `>>` instead of `>` to avoid overwriting the file instead of adding lines.

Comment (knusbaum) [reply utils/scripts/load-binary.sh:224]: Doh! You're exactly right.

**Review by darccio — APPROVED** [review]

**Comment by darccio** [comment]
Oops, fixing.

**Comment by darccio** [comment]
@knusbaum I accidently merged main into this PR. I reverted that and recovered your changes, including the one from my suggestion. I'll do a quick review to make sure everything is fine.

**Comment by darccio** [comment]
Thanks @knusbaum!

**Comment by cbeauchesne** [comment]
@darccio I'm a little bit worried to see a file impacting all parametric scenrio modified, and 5 parametric job failing on the last commit. Are you 100% sure it's not related?

**Comment by cbeauchesne** [comment]
So, my bad, you didn't merged it in `main`. I need some rest 😅

**Pull request merged**